core/src/main/java/org/apache/accumulo/core/fate/zookeeper/FateLock.java

```diff
@@ -68,16 +74,56 @@ public FateLock(ZooReaderWriter zrw, FateLockPath path) {
     this.path = requireNonNull(path);
   }
 
+  public static class FateLockNode implements Comparable<FateLockNode> {
+    public final long sequence;
+    public final String lockData;
+
+    private FateLockNode(String nodeName) {
+      int len = nodeName.length();
+      Preconditions.checkArgument(nodeName.startsWith(PREFIX) && nodeName.charAt(len - 11) == '#',
+          "Illegal node name %s", nodeName);
+      sequence = Long.parseLong(nodeName.substring(len - 10));
+      lockData = nodeName.substring(PREFIX.length(), len - 11);
```

Comment:
So, while this is probably unlikely to happen, the sequential numbering can overflow causing a negative to occur, which I think would cause a problem with this parsing instead of splitting on `#`.

Reply:
Added handing and test for negative seq numbers in 2309b9b